deploy/config/crds/submarineraddon.open-cluster-management.io_submarinerdiagnoseconfigs.yaml

```diff
@@ -0,0 +1,174 @@
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.8.0
+  creationTimestamp: null
+  name: submarinerdiagnoseconfigs.submarineraddon.open-cluster-management.io
+spec:
+  group: submarineraddon.open-cluster-management.io
+  names:
+    kind: SubmarinerDiagnoseConfig
+    listKind: SubmarinerDiagnoseConfigList
+    plural: submarinerdiagnoseconfigs
+    singular: submarinerdiagnoseconfig
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: SubmarinerDiagnoseConfig represents the configuration for SubmarinerDiagnose.
+          Submariner-addon will use it to run SubmarinerDiagnose.
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: Spec defines the configuration of the Submariner
+            properties:
+              CNI:
+                type: boolean
+              all:
+                type: boolean
+              connections:
+                type: boolean
+              deployment:
+                type: boolean
+              firewall:
+                type: boolean
+              firewallOptions:
+                properties:
+                  interCluster:
+                    type: boolean
+                  intraCluster:
+                    type: boolean
+                  metrics:
+                    type: boolean
+                  remoteCluster:
+                    type: string
+                  remoteK8sAPIServer:
+                    type: string
+                  remoteK8sAPIServerToken:
+                    type: string
+                  remoteK8sCA:
+                    type: string
+                  remoteK8sRemoteNamespace:
+                    type: string
+                  remoteK8sSecret:
+                    type: string
+                type: object
+              gatherLogs:
```

Comment:
what does gatherLogs do?

Reply:
Nothing yet. This is for a possible future use case where we may want to reuse this to gather logs too.